src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.service.impl;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final int MIN_PASSWORD_LENGTH = 8;
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = userService.findByEmail(email)
+                .orElseThrow(() -> new AuthenticationException("Please "
+                        + "enter correct email or password"));
+        if (user.getPassword().equals(HashUtil.hashPassword(password, user.getSalt()))) {
+            return user;
+        }
+        throw new AuthenticationException("Please enter correct email or password");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (password.length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password less than 8 symbols");
+        }
+        if (password.contains("[/^[!@#$%^&*()_+\\-=\\[\\]{};':\"\\\\|,.<>\\/?]*$/]")) {
+            throw new RegistrationException("Password must contain only letters or figures,"
+                    + "any special symbols, please");
```

Comment:
It seems should be like this?
```suggestion
            throw new RegistrationException("Password must contain only letters or figures,"
                    + "avoid any special symbols, please");
```

Reply:
), 100%